cocos/2d/CCTMXObjectGroup.h

```diff
@@ -31,24 +31,116 @@ THE SOFTWARE.
 #include "math/CCGeometry.h"
 #include "base/CCValue.h"
 #include "base/CCRef.h"
+#include "2d/CCSprite.h"
+#include "2d/CCDrawNode.h"
 
 NS_CC_BEGIN
 
+typedef enum TMXObjectType_ {
+    TMXObjectTypeRect,
+    TMXObjectTypeEllipse,
+    TMXObjectTypePolygon,
+    TMXObjectTypePolyline,
+    TMXObjectTypeImage
+} TMXObjectType;
```

Comment:
How about using `enum class` which safer. It will look like:
```c++
enum class TMXObjectType
{
    RECT,
    ELLIPSE,
    POLYGON,
    POLYLINE,
    IMAGE
} TMXObjectType;
```